### PR TITLE
Document vault.process

### PR DESCRIPTION
**Pull request by marcusolsson**

### CODE REVIEW

File: en/Plugins/Vault.md [hunk 1]
```diff
@@ -55,6 +55,39 @@ export default class ExamplePlugin extends Plugin {
...
+}
+```
+
+If you want to modify a file based on its current content, use [[process|Vault.process()]] instead. The second argument is a callback that provides the current file content and returns the modified content.
+
+```ts
+// emojify replaces all occurrences of :) with 🙂.
+function emojify(vault: Vault, file: TFile): Promise<string> {
+  return vault.process(file, (data) => {
+    return data.replace(":)", 🙂);
```
Comment (lishid) [inline en/Plugins/Vault.md:74]: Shouldn't the emoji also be quoted?

File: en/Plugins/Vault.md [hunk 2]
```diff
@@ -55,6 +55,39 @@ export default class ExamplePlugin extends Plugin {
...
+
+### Asynchronous modifications
+
+[[process|Vault.process()]] only supports synchronous modifications. If you need to modify a file asynchronously:
+
+1. Read the file using [[cachedRead|Vault.cachedRead()]].
+2. Perform the async operations.
+3. Update the file using [[Reference/TypeScript API/Vault/process|Vault.process()]].
+
+Remember to check that the `data` in the `process()` callback is the same as the data returned by `cachedRead()`. If they aren't the same, that means that the file was changed by a different process, and you may want to ask the user for confirmation.
```
Comment (lishid) [inline en/Plugins/Vault.md:89]: Should add to the end "or try again".

**Review by lishid — APPROVED** [review]

File: en/Plugins/Vault.md (hunk 1 above)
Comment (marcusolsson) [reply en/Plugins/Vault.md:74]:
```suggestion
    return data.replace(":)", "🙂");
```

Good catch!

File: en/Plugins/Vault.md (hunk 2 above)
Comment (marcusolsson) [reply en/Plugins/Vault.md:89]:
```suggestion
Remember to check that the `data` in the `process()` callback is the same as the data returned by `cachedRead()`. If they aren't the same, that means that the file was changed by a different process, and you may want to ask the user for confirmation, or try again.
```

**Review by lishid — APPROVED** [review]

**Pull request merged**